app/models/data_resources/event_record.rb

```diff
@@ -89,6 +93,25 @@ class EventRecord < ApplicationRecord
       .distinct
   }
 
+  scope :upcoming_with_date_select, lambda {
+    select_clause = <<~SQL
+      event_records.*,
+      fixed_dates.id AS fixed_date_id,
+      fixed_dates.date_start AS fixed_date_start,
+      fixed_dates.date_end AS fixed_date_end,
+      fixed_dates.time_start AS fixed_time_start,
+      fixed_dates.time_end AS fixed_time_end,
+      fixed_dates.weekday AS fixed_weekday,
+      fixed_dates.time_description AS fixed_time_description,
+      fixed_dates.use_only_time_description AS fixed_use_only_time_description
+    SQL
+
+    # ignore the first date for recurring events, because it is the original date object with
+    # a time span that should not be listed in the returning event records.
+    where("fixed_dates.id != (SELECT MIN(fd.id) FROM fixed_dates fd WHERE fd.dateable_id = event_records.id)")
+      .select(select_clause)
+  }
```

Comment:
this is new and return the event records the way we need. sadly the loading time increased with this. sadly I could not find a way to make this fast. in my local data base with three events including one recurring event that has a very large time span the loading time is more than 10 seconds. any ideas?